### PR TITLE
Linux kernel 5.15.22

### PR DESCRIPTION
**Pull request by MaxKellermann**
Updates to the latest Linux 5.10.y kernel. Next time, I'll update to the latest LTS kernel 5.15.y.

By disabling SATA support, this saves 200ms kernel boot time. Does anybody use SATA on an OpenVario?

New record: power on to ovmenu in 11.3 seconds.

### CODE REVIEW

**Comment by MaxKellermann** [comment]
> Next time, I'll update to the latest LTS kernel 5.15.y.

This is ... "next time". Updated to 5.15.22 in a rather trivial commit.
The massive cleanup that precedes it was the difficult part ...

**Comment by linuxianer99** [comment]
> Updates to the latest Linux 5.10.y kernel. Next time, I'll update to the latest LTS kernel 5.15.y.
> 
> By disabling SATA support, this saves 200ms kernel boot time. Does anybody use SATA on an OpenVario?
> 
> New record: power on to ovmenu in 11.3 seconds.

Think SATA is never used.

Great boot time !! Congratulations !!

**Review by linuxianer99 — COMMENTED** [review]
Looks good !
Just one question: Why are we going back to a "own" kernel recipe instead of use the bbappend ??

**Comment by MaxKellermann** [comment]
> Why are we going back to a "own" kernel recipe instead of use the bbappend ??

The sunxi recipe is stuck with 5.10.2, and there's no perspective this will change. Also the sunxi recipe is full of useless stuff.
Then there's the linux-yocto recipe, also full of useless stuff.
And now, look at how simple the new recipe is. It inherits from "kernel.bbclass" and "kernel-yocto.bbclass", and contains just what is special for OpenVario. A bbappend would not be simpler. By inheriting from those bbclasses, we are better than "bbappending".

**Review by linuxianer99 — APPROVED** [review]

**Comment by linuxianer99** [comment]
> > Why are we going back to a "own" kernel recipe instead of use the bbappend ??
> 
> The sunxi recipe is stuck with 5.10.2, and there's no perspective this will change. Also the sunxi recipe is full of useless stuff. Then there's the linux-yocto recipe, also full of useless stuff. And now, look at how simple the new recipe is. It inherits from "kernel.bbclass" and "kernel-yocto.bbclass", and contains just what is special for OpenVario. A bbappend would not be simpler. By inheriting from those bbclasses, we are better than "bbappending".

got it! Thanks for explaination

**Comment by MaxKellermann** [comment]
Linux 5.15.23 was just released - I amended the patch.

**Comment by MaxKellermann** [comment]
Amended one patch to remove an obsolete variable, and added another patch to remove `DEFAULTTUNE`. Not exactly related to the kernel, but doesn't belong anywhere else ...

File: meta-ov/recipes-kernel/linux/files/openvario.cfg [hunk 1]
```diff
@@ -25,7 +25,6 @@ CONFIG_UEVENT_HELPER=y
 CONFIG_INPUT_EVDEV=y
 # CONFIG_INPUT_FF_MEMLESS is not set
 # CONFIG_INPUT_MOUSE is not set
-CONFIG_INPUT_UINPUT=y
```
Comment (kedder) [inline meta-ov/recipes-kernel/linux/files/openvario.cfg:30]: I think touchscreen handling is based on uinput (`ts_uinput` daemon). If it is the same uinput, suggest to leave it enabled.

**Review by kedder — CHANGES_REQUESTED** [review]

File: meta-ov/recipes-kernel/linux/files/openvario.cfg (hunk 1 above)
Comment (MaxKellermann) [reply meta-ov/recipes-kernel/linux/files/openvario.cfg:30]: It is still enabled; the line was duplicate, and this commit just removes one copy.

**Review by kedder — APPROVED** [review]

**Pull request merged**